client/src/components/PopupDialog.vue

```diff
@@ -0,0 +1,49 @@
+<script setup lang="ts">
+import { computed } from 'vue'
+import { useDisplay } from 'vuetify'
+
+const emit = defineEmits(['update:modelValue', 'handleSubmit'])
+const props = defineProps(['modelValue', 'title', 'text', 'submitText'])
+
+const value = computed({
+  get() {
+    return props.modelValue
+  },
+  set(value) {
+    emit('update:modelValue', value)
+  }
+})
+
+const { mobile } = useDisplay()
+</script>
+
+<template>
+  <v-dialog v-model="value" :fullscreen="mobile" max-width="912px">
```

Comment:
We dont need this to be fullscreen for mobile - its just a small popup so it should be fine